tests/unit/output/test_local.py

```diff
@@ -1,25 +1,48 @@
-from mock import patch
+import os
 
+from mock import patch
 from dvc.output import OutputLOCAL
 from dvc.remote.local import RemoteLOCAL
 from dvc.stage import Stage
+from dvc.utils import relpath
 from tests.basic_env import TestDvc
 
 
 class TestOutputLOCAL(TestDvc):
     def _get_cls(self):
         return OutputLOCAL
 
-    def _get_output(self):
-        stage = Stage(self.dvc)
-        return self._get_cls()(stage, "path", cache=False)
+    def _get_output(self, wdir=None, path="path"):
+        stage = Stage(self.dvc, wdir=wdir if wdir else os.curdir)
+        return self._get_cls()(stage, path, cache=False)
 
     def test_save_missing(self):
         o = self._get_output()
         with patch.object(o.remote, "exists", return_value=False):
             with self.assertRaises(o.DoesNotExistError):
                 o.save()
 
+    def test_str_if_workdir_is_inside_repo(self):
+        work_dir = os.path.join(self.root_dir, "models")
+        os.mkdir(work_dir)
+        os.chdir(work_dir)
+
+        o = self._get_output(wdir=work_dir)
+        self.assertEqual("path", str(o))
+
+    def test_str_if_workdir_outside_repo(self):
```

Comment:
@efiop, this test is failing for `SSH`/`HDFS`/`s3`/`GS` as they are remote. Others are passing.
Should I just separate this test (and, keep it only for local)?

Reply:
You may separate `TestOutputBASE` and descend from that, not from local one. Add specific local tests to `TestOutputLocal`.

Ultimately we are switching to pytest-style function based tests, but this is to big to include it here I guess. Even for output tests only.

Reply:
I have separated the test for now. Tell me, I feel, I have done something wrong here :smiley:

Reply:
Now the three old tests are not tested for other remotes than local.

Reply:
Considering all the test complications I suggest not touching existing ones, but adding a separate pytest-style test to check output stratification for an output in external repo.

Reply:
> Now the three old tests are not tested for other remotes than local.
They do run, as the above `TestCase` is not changed at all. Or, did you mean this specific example?

> Considering all the test complications I suggest not touching existing ones, but adding a separate pytest-style test to check output stringification for an output in external repo.
You mean, for each remote separately? Please, clarify.

Reply:
@skshetry @Suor is saying that you've modified TestOutputLOCAL which is used in tests for other remotes, hence why the tests are failing and why not all tests are now running for those remotes. You need to not touch the old tests, but create a new one pytest-style test to test that stringification works properly for local output when you are in and outside of the repo.